File: app/src/main/java/com/github/weatherapp/ui/resource/MessageResourceProvider.java
package com.github.weatherapp.ui.resource;

import com.github.weatherapp.core.CurrentForecast;

public interface MessageResourceProvider {
    String getCityDescription(CurrentForecast forecast);

    String getTemperatureDescription(CurrentForecast forecast);

    String getMetricUnitDescription(String metricUnit);
}

File: app-kotlin/src/main/kotlin/com/github/weatherapp/ui/resource/MessageResourceProvider.kt
package com.github.weatherapp.ui.resource

import com.github.weatherapp.core.CurrentForecast

interface MessageResourceProvider {
    fun getCityDescription(forecast: CurrentForecast): String

    fun getTemperatureDescription(forecast: CurrentForecast): String

    fun getMetricUnitDescription(metricUnit: String): String
}
